scripts/fast-start.sh

```diff
@@ -100,6 +100,8 @@ fi
 # Let's try to detect the environment we are using.
 PUBLIC_IP=""
 PRIVATE_IP=""
+AZURE_METADATA=$(curl -s -o /dev/null -w "%{http_code}" http://169.254.169.254/metadata/v1/InstanceInfo)
```

Comment:
I'd move this up at the top where we define the Google Metadata service and keep it just as the endpoint.
You can do the curl command straight up in the check block. That way it follows the pattern for the rest of providers.